docker-compose.yml

```diff
@@ -0,0 +1,58 @@
+version: '3.8'
+
+# Ports
+# Builder: 3000
+# Explorer: 3001
+# Manager: 3002
+# Pinata: 3003
+# Indexer: 3004
+# Dev chain 1: 3005
+# Subgraph dev chain 1: 3006
+# Dev chain 2: 3007
+# Subgraph dev chain 2: 3008
+
+services:
+  pinata:
+    image: ghcr.io/gitcoinco/pina:init
+    ports:
+      - '127.0.0.1:3003:8000'
+
+  localchain-1:
+    image: ghcr.io/gitcoinco/allo-contracts-v1:pr-1
+    ports:
+      - '127.0.0.1:3005:8545'
+    environment:
+      - DEV_CHAIN_ID=313371
+
+  localchain-1-init:
```

Comment:
what is the difference between the 3 `localchain` services here, my guess is one is the service and the other one is initializing something, but i don't get why they're separate if you don't mind explaining

Reply:
@boudra sure, for each chain we have 2 things, `localchain` is the actual chain running, and `localchain-init` is the one that populates it. 
At the beginning I though that we could populate the chain (deploy contracts and create projects/rounds) at image build time. `anvil` allows to save the state of the chain, so my idea was then to run the container loading the same state for the chain. 
But `anvil` doesn't allow to dump events, it dumps only the chain state. And both for the indexer and Builder we need all the events. 
One idea could also be to run the container and then `docker exec` to populate, but having everything in the dockerfile felt better to have it automated. Is there a better way to do it?
Another thing to keep in mind is that when we populate we need the pinata (`pina`) service running to save all the metadata, so the container needs to be linked.

Reply:
Maybe just a matter of renaming it `localchain-1-initIALIZER` to remove doubts? @boudra would that have made it clearer? Or a few comments in the source?

Reply:
@bard @boudra good idea, I renamed it and added a comment

Reply:
yeah a comment is perfect, I like to approach, it was more to confirm my assumption